user_management/api/views.py

```diff
@@ -151,7 +152,8 @@ def initial(self, request, *args, **kwargs):
         try:
             self.user = User.objects.get(pk=uid)
         except User.DoesNotExist:
-            raise Http404()
+            msg = _('Invalid or expired token.')
+            raise PermissionDenied(detail=msg)
```

Comment:
Also see 419:
> Not a part of the HTTP standard, 419 Authentication Timeout denotes that previously valid authentication has expired. It is used as an alternative to 401 Unauthorized in order to differentiate from otherwise authenticated clients being denied access to specific server resources.